cvmfs/telemetry_aggregator.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * This file is part of the CernVM File System.
+ *
+ * TelemetryAggregator class is a thread that snapshots the internal counters
+ * of cvmfs statistic object. A custom telemetry class is needed to perform
+ * the step of manipulating and sending/storing the counters.
+ */
+
+#ifndef CVMFS_TELEMETRY_AGGREGATOR_H_
+#define CVMFS_TELEMETRY_AGGREGATOR_H_
+
+#include <pthread.h>
+#include <stdint.h>
+
+#include <cstring>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "options.h"
+#include "statistics.h"
+#include "util/single_copy.h"
+
+namespace perf {
+
+// Errors of telemetry classes (including custom classes)
+enum TelemetryError {
+  kTelemetrySuccess = 0,
+  kTelemetryHostAddress,
```

Comment:
Same for the ones below, let's indicate that these are error codes.

```suggestion
  kTelemetryFailHostAddress,
```